fido2-net-lib/EnumExtensions.cs

```diff
@@ -17,14 +18,24 @@ public static class EnumExtensions
         /// <exception cref="System.ArgumentException">No XmlEnumAttribute code exists for type " + typeof(TEnum).ToString() + " corresponding to value of " + value</exception>
         public static TEnum ToEnum<TEnum>(this string value, bool ignoreCase = true) where TEnum : struct, Enum
         {
-            if (Enum.TryParse<TEnum>(value, ignoreCase, out var enumValue))
-            {
-                return enumValue;
-            }
-            else 
+            // Try to parse it normally on the first try
+            if (Enum.TryParse<TEnum>(value, ignoreCase, out var result))
+                return result;
+
+            // Try with value from EnumMemberAttribute
+            foreach (var o in Enum.GetValues(typeof(TEnum)))
```

Comment:
We could reuse the subsquent usage of `Enum.GetValues(typeof(TEnum))` on line 34 and 35.
But this is really minor, no reason to not merge it as is.

Reply:
I also thought about it but wanted to avoid allocating the valid values, thinking it will be less common to fall into the exception and the allocation will be of no use when the loop returns the value.

Reply:
Hmm, not sure I follow. But as soon you reach the `foreach` you allocating the valid values.
I didn't try that, but this should work:

```
            // Try with value from EnumMemberAttribute
            var values = Enum.GetValues(typeof(TEnum)).OfType<TEnum>().ToArray();
            foreach (var o in values)
            {
                if (ToEnumMemberValue(o).Equals(value, ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))
                    return o;
            }

             throw new ArgumentException($"Value '{value}' is not a valid enum name of '{typeof(TEnum)}' ({nameof(ignoreCase)}={ignoreCase}). Valid values are: {string.Join(", ", values))}.");
        }
```

Reply:
The returned valid values are converted using `ToEnumMemberValue` in order to spill out the strings like *cross-platform*.

Reply:
Ah crap, yes sure. Then the solution would be:

     ```
           // Try with value from EnumMemberAttribute
            var values = Enum.GetValues(typeof(TEnum)).OfType<TEnum>().ToArray();
            foreach (var o in values)
            {
                if (ToEnumMemberValue(o).Equals(value, ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))
                    return o;
            }

             throw new ArgumentException($"Value '{value}' is not a valid enum name of '{typeof(TEnum)}' ({nameof(ignoreCase)}={ignoreCase}). Valid values are: {string.Join(", ", values.Select(v => ToEnumMemberValue(n)))}.");
```
But as I already said, I'm fine with what you did.

Reply:
The LINQ is internally doing a loop and allocating a new collection, but I have to agree that it looks more verbose and cleaner that way. 🙂 I will make the proper changes as suggested. Thanks!